src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,86 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
 
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (interfaceClazz.isAnnotationPresent(Component.class)) {
+            Class<?> clazz = findImplementation(interfaceClazz);
```

Comment:
Flip this `if` to follow the return/fail early pattern and not overcomplicate `if/else`s
```suggestion
        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
            throw new RuntimeException( // replace it with some custom one
                    "Injection failed, missing @Component annotation on the class "
                            + interfaceClazz.getName());
        }
        Class<?> clazz = findImplementation(interfaceClazz);
        // other main logic
```